### DISCUSSION TITLE
Thermal analysis for devices without thermal models like legacy BJTs

### DISCUSSION THREAD

**Discussion by tomhajjar** [ra3xdh/qucs_s#1020] [Show and tell]
I based this project on a 1994 EDN Design Note. The Author used Spice sub-circuits to measure the power dissipation of Audio power amp BJT output devices. I replicated his work in Qucs-S. This same idea can be extended to almost any device. 

The Author is still active on EEVblog. They guys at LTspice forum also helped.

The accuracy of the simulation is limited by the accuracy of the Electrothermal device parameters and how many "R/C" stages are used. Every "transition" between the Device Die and the heatsink should be accounted for like Die bond, lead frame, thermal compound, insulator...etc.

![2024-10-22_124616](https://github.com/user-attachments/assets/5f7ee87b-79d4-405b-9d40-33e4e3c05f59)
![2024-10-24_170138](https://github.com/user-attachments/assets/e784177f-7f30-4b76-9bd1-319ceaad0cc3)
![2024-10-24_170202](https://github.com/user-attachments/assets/f6e97e3b-cf02-4eca-b366-fcea6c27f709)

[Amp_BJT_thermal_prj.zip](https://github.com/user-attachments/files/17512950/Amp_BJT_thermal_prj.zip)

**Comment by ivandi69** [comment]
This gives acceptable result with much less efforts:
![image](https://github.com/user-attachments/assets/f16ac7a0-28a0-4300-8e73-7b3ad5ac2421)
[BJT_output_thermal_2.sch.txt](https://github.com/user-attachments/files/17612026/BJT_output_thermal_2.sch.txt)

**Comment by tomhajjar** [comment]
Nice work. Our results are different @T=10sec. I assume because the 2 currents don't sum at Rha.

Eliminating all the Cth ignores thermal time constants. Your "curve" is flat. For a power amp running continuously at high levels, this is moot. For circuits with variable duty cycles, Cth might matter.

![2024-11-03_162222](https://github.com/user-attachments/assets/6d447f4f-b506-41f5-ba7a-18b0c4048152)
![2024-11-03_162157](https://github.com/user-attachments/assets/e79b676f-7fff-44dc-8b6f-314df65ed828)

[BJT_thermal_prj.zip](https://github.com/user-attachments/files/17612114/BJT_thermal_prj.zip)

**Comment by ivandi69** [comment]
Here is another way to do the same thing. Unfortunately the B-source in ngspice doesn't accept expressions like q1:power or p(q1).
![image](https://github.com/user-attachments/assets/60b51b6d-aace-41c8-b8d0-66c9387fd0a8)
[BJT_output_thermal_3.sch.txt](https://github.com/user-attachments/files/17625424/BJT_output_thermal_3.sch.txt)

**Comment by tomhajjar** [comment]
Very nice. Both your solutions are viable. The duty cycle of the circuit would determine the choice.

The goal of the project is to expose Users to Thermoelectric simulations. It is up to the User to determine how many R/C sections to use and what the R/C parameters are. Every thermal "barrier" requires an RC section be added. For example thermal insulating pads and heatsink compound are a barrier. 

I have had lot's of issues getting Manufacturer's thermal models to work in ngspice but almost every model works in LTspice. These techniques bypass compatibility issues plus allow any device to be analyzed.

I didn't realize this is the same as Holger's method...
https://ngspice.sourceforge.io/ngspice-electrothermal-tutorial.html

[BJT_output_thermal_3.zip](https://github.com/user-attachments/files/17626321/BJT_output_thermal_3.zip)
